### PR TITLE
[CI] Adding MPI meshing python tests

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
This adds the MPI meshing app suite to the CI.

It is currently empty, PR #7570  will add the first test.

**Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- Created MPI test suite
- Added to CI

### CODE REVIEW

**Comment by philbucher** [comment]
I will take a look soon, please wait with merging, I have some comments

File: applications/MeshingApplication/tests/test_MeshingApplication_mpi.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+try:
+    import KratosMultiphysics.mpi as KratosMPI
+    import KratosMultiphysics.MetisApplication as MetisApplication
+    import KratosMultiphysics.TrilinosApplication as TrilinosApplication
+except ImportError:
+    raise Exception("KratosMPI could not be imported!")
+
+if KratosMultiphysics.ParallelEnvironment.GetDefaultSize() != 2:
+    raise Exception("The MPI tests currently support only being run with 2 processors!")
```
Comment (philbucher) [inline applications/MeshingApplication/tests/test_MeshingApplication_mpi.py]:
```suggestion
```

this should not be here any more, please check `test_CoSimulationApplication_mpi.py`, this is the most up to date one

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -219,6 +219,13 @@ jobs:
         export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
         mpiexec --oversubscribe -np 4 python3 applications/ParticleMechanicsApplication/tests/test_ParticleMechanicsApplication_mpi.py --using-mpi
 
+    - name: Running MPI Python MeshingApplication (2 Cores)
+      run: |
+        . /opt/intel/mkl/bin/mklvars.sh
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        mpiexec -np 2 python3 applications/MeshingApplication/tests/test_MeshingApplication_mpi.py --using-mpi
+
```
Comment (philbucher) [inline .github/workflows/ci.yml:228]: please also add it for 3 and 4 cores

**Review by philbucher — COMMENTED** [review]
minor, seems like you used some old stuff (my bad I haven't updated yet :D )

File: applications/MeshingApplication/tests/test_MeshingApplication_mpi.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    nightlyMPISuite.addTests(smallMPISuite)
+
+    ### Full MPI set ###########################################################
+    allMPISuite = suites['mpi_all']
+    allMPISuite.addTests(nightlyMPISuite) # already contains the smallMPISuite
+
+    allSuite = suites['all']
+    allSuite.addTests(allMPISuite)
+
+    return suites
```
Comment (philbucher) [inline applications/MeshingApplication/tests/test_MeshingApplication_mpi.py:37]: wait, you are not adding any tests? :sweat_smile:

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -219,6 +219,27 @@ jobs:
...
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        mpiexec -np 3 python3 applications/MeshingApplication/tests/test_MeshingApplication_mpi.py --using-mpi
+
+    - name: Running MPI Python MeshingApplication (4 Cores)
+      run: |
+        . /opt/intel/mkl/bin/mklvars.sh
+        export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
+        export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        mpiexec -np 4 python3 applications/MeshingApplication/tests/test_MeshingApplication_mpi.py --using-mpi
```
Comment (philbucher) [inline .github/workflows/ci.yml]: check the other tests, you need the `oversubscribe` option

**Review by philbucher — COMMENTED** [review]
more commens

File: .github/workflows/ci.yml (hunk 4 above)
Comment (philbucher) [reply .github/workflows/ci.yml]: (thats the the CI crashed)

Comment (marcnunezc) [reply .github/workflows/ci.yml]: Oops ok!

File: applications/MeshingApplication/tests/test_MeshingApplication_mpi.py (hunk 3 above)
Comment (marcnunezc) [reply applications/MeshingApplication/tests/test_MeshingApplication_mpi.py:37]: No, there are no tests here! Im adding one in #7570, but I thought it was clearer if I did the CI change in an different PR. (so the tests in #7570 can appear in the CI)

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**